packages/components/socialFeed/NewsFeed/MapComponent/AddressSearch.tsx

```diff
@@ -0,0 +1,157 @@
+import { useEffect, useState } from "react";
+import {
+  ActivityIndicator,
+  ScrollView,
+  TouchableOpacity,
+  View,
+  ViewStyle,
+} from "react-native";
+
+import location from "../../../../../assets/icons/location.svg";
+import {
+  neutral77,
+  primaryColor,
+  secondaryColor,
+} from "../../../../utils/style/colors";
+import { fontSemibold13 } from "../../../../utils/style/fonts";
+import { BrandText } from "../../../BrandText";
+import { SVG } from "../../../SVG";
+import { TextInputCustom } from "../../../inputs/TextInputCustom";
+import { TextInputOutsideLabel } from "../../../inputs/TextInputOutsideLabel";
+
+let typingTimeout: NodeJS.Timeout | undefined;
+
+interface TAddressSearchProps {
+  addressPlaceHolder: any;
+  address: any;
+  setAddress: any;
+  setLocationSelected: any;
+  setAddressPlaceHolder: any;
+}
+
+export const AddressSearch: React.FC<TAddressSearchProps> = ({
+  addressPlaceHolder,
+  address,
+  setAddress,
+  setLocationSelected,
+  setAddressPlaceHolder,
+}) => {
+  const [results, setResults] = useState<any>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const searchAddress = async (params: string) => {
+    return new Promise((resolve, reject) => {
```

Comment:
maybe it makes sense to just use react query, it will better match the rest of the project